Release 1.32.0 doc changes (#102)
nginx/unit-docs: source/howto/source.rst

.. nxt_details:: Enabling WebAssembly
   :hash: source-wasm

   .. tabs::
      :prefix: source-enable-webassembly
      :toc:

      .. tab:: wasm-wasi-component

          To build Unit with support for the WebAssembly Component Model, you need **rust**
          version 1.76.0+, **cargo** and the developer package for **clang** as mentioned in the :ref:`Required Software Section <source-prereq-build>`.

          Next please refer to :ref:`Configuring Modules - WebAssembly <modules-webassembly>` for further instructions.

      .. tab:: unit-wasm

          .. warning::
             Unit 1.32 and later support the WebAssembly Component Model and WASI 0.2 APIs.

>              We recommend to use the new implementation.

Reviewer: ```suggestion
             We recommend using the new implementation.
```

          To build Unit with the `WebAssembly <https://webassembly.org>`__
          language module,
          you need the
          `Wasmtime <https://wasmtime.dev>`__
          runtime.
          Download it C API
          `files <https://github.com/bytecodealliance/wasmtime/releases/>`__
          suitable for your OS and architecture
          to the same parent directory
          as the Unit code,
          for example:

          .. code-block:: console

             $ cd ..
             $ wget -O- https://github.com/bytecodealliance/wasmtime/releases/download/v12.0.0/wasmtime-v12.0.0-x86_64-linux-c-api.tar.xz \
                   | tar Jxf -  # Unpacks to the current directory

          Point to the resulting **include** and **lib** directories when
          :ref:`configuring <howto/source-modules-webassembly>` the Unit code.

>           To build WebAssembly apps that run on Unit, you will also need

Reviewer: ```suggestion
          To build WebAssembly apps that run on Unit, you need
```

.. nxt_details:: Enabling WebAssembly
   :hash: source-wasm

   .. tabs::
      :prefix: source-enable-webassembly
      :toc:

      .. tab:: wasm-wasi-component

          To build Unit with support for the WebAssembly Component Model,
          you need **rust** version 1.76.0+, **cargo** and the developer
          package for **clang** as mentioned in the
          :ref:`Required Software Section <source-prereq-build>`.

          Next please refer to
          :ref:`Configuring Modules - WebAssembly <modules-webassembly>` for
          further instructions.

      .. tab:: unit-wasm

          .. warning::

>              Unit 1.32 and later support the WebAssembly Component Model
>              and WASI 0.2 APIs.
>              We recommend using the new implementation.

Reviewer 2: I don't find these warnings (there are several in the PR) sufficiently clear. To grok this, I have to already understand what the Component Model and WASI 0.2 APIs are. I also have to somehow infer that the thing I'm looking at, `unit-wasm`, is _not_ that. And I have to then try to figure out wtf the "new implementation" is.

Maybe something like this would be better:

```suggestion
             The `unit-wasm` module is deprecated.
             We recommend using `wasm-wasi-component` instead, which supports
             WebAssembly Components using standard WASI 0.2 interfaces.
             The `wasm-wasi-component` module is available in Unit 1.32 and later.
```

Probably not perfect, but maybe a little more instructive?

Author: ```suggestion
             The **unit-wasm** module is deprecated.
             We recommend using **wasm-wasi-component** instead,
             available in Unit 1.32.0 and later, which supports 
             WebAssembly Components using standard WASI 0.2 interfaces.
```

          To build Unit with the `WebAssembly <https://webassembly.org>`__
          language module,
          you need the
          `Wasmtime <https://wasmtime.dev>`__
          runtime.

>           Download it C API

Reviewer 3: I guess at the very least that should be s/it/the/

   .. tab:: WebAssembly

      .. _modules-webassembly:

      When you run :program:`./configure wasm-wasi-component`,
      the script configures a module to support running WebAssembly applications on Unit.

   .. tab:: Unit-Wasm

      .. warning::
         Unit 1.32 and later support the WebAssembly Component Model and WASI 0.2 APIs.

>          We recommend to use the new implementation.

Reviewer: ```suggestion
         We recommend using the new implementation.
```

   .. tab:: WebAssembly

      .. _modules-webassembly:

      When you run :program:`./configure wasm-wasi-component`,
      the script configures a module to support running WebAssembly

>       components on Unit.

Reviewer 4: When you run :program:`./configure wasm-wasi-component`,
the script configures a module to support running WebAssembly
components on Unit.

The language module doesn't accept any extra configuration parameters. The module's basename will always be **wasm-wasi-component** and can not be changed.

Reviewer 5: ```suggestion
      When you run :program:./configure wasm-wasi-component,
      the script configures a module to support running WebAssembly
      components on Unit.

      The module doesn't accept any extra configuration parameters.
      The module's basename is wasm-wasi-component.
```

Author: ```suggestion
      When you run :program:`./configure wasm-wasi-component`,
      the script configures a module to support running WebAssembly
      components on Unit.
      
      The module's basename is **wasm-wasi-component** and does
      not accept any extra configuration parameters.
      ```

Reviewer 5: The conjunction is strange because there are different subjects. The basename relates to the module itself, where are configuration parameters relates to the `configure` command.

>       When you run :program:`./configure wasm-wasi-component`,
>       the script configures a module to support running WebAssembly
>       applications on Unit.

Author: ```suggestion
      When you run :program:`./configure wasm-wasi-component`,
      the script configures a module to support running WebAssembly
      components on Unit.
```

   .. tab:: Unit-Wasm

      .. warning::

>          Unit 1.32 and later support the WebAssembly Component Model and WASI

Reviewer 3: `1.32.0`

Author: ```suggestion
         Unit 1.32.0 and later support the WebAssembly Component Model and WASI
```
